### PR TITLE
PDX-508: Port render/load-blocking check types to local validator

### PR DESCRIPTION
**Pull request by mrdailey99**
## Summary
- Implements the eight render / load-blocking check types that were referenced by rules but **unregistered**, so the local validator silently skipped them and diverged from the Quality Hub score.
- Faithful ports of the Quality Hub validators, preserving score parity with the Lambda (one violation per rule; `count` only when >1; the casing checks cap at 5).

| Check type | Rule (severity) | Fires when |
|---|---|---|
| `valueClassCasing` | RENDER-CASE-001 (critical) | known `valueClass` spelled wrong-case (`Boolean`→`boolean`); unknown class doesn't fire |
| `booleanCasing` | RENDER-BOOL-001 (critical) | `<value valueClass="boolean">` text is `True/False/TRUE/FALSE` |
| `invalidValueClass` | VALUE-CLASS-001 (critical) | `class` not a valid Provar value class, or `class="value"` w/ bad `valueClass` |
| `dateValueClassFormat` | RENDER-DATE-VALUECLASS-001 (critical) | `valueClass="date"/"dateTime"` text is an ISO string, not epoch millis |
| `setValuesInvalidElements` | SETVALUES-INVALID-ELEMENT-001 (critical) | `SetValues` has `<namedValueSet>`/`<name>` or bad `<namedValues>` child |
| `apexConnectValidArguments` | APEX-CONNECT-ARGS-001 (critical) | `ApexConnect` argument id outside the 20-id whitelist |
| `apexConnectConnectionIdValueClass` | APEX-CONNECT-CONNID-001 (critical) | `connectionId` value uses `valueClass="string"`/non-`id` |
| `apexConnectReuseConnection` | APEX-REUSE-CONN-001 (major) | `reuseConnectionName` carries a value (must be blank) |

This is **scope item 2** of the PDX-508 umbrella (load-blocking check types). Item 1 (`mustContainArgument`, PR #207) shipped earlier; tiers 3–6 follow in separate PRs.

## Backend parity notes (validated against `quality-hub-agents` `best_practices_engine.py`)
- `valueClassCasing` preserves the back-end quirk where `funcCall`/`valueList` casing never fires (their lowercased form isn't in the valid set).
- `invalidValueClass` mirrors the back-end's **hardcoded** `VALID_CLASSES` (which accepts `invalid`/`namedValues` and the camelCase `dateTime`) — it intentionally ignores the rule JSON's `validClasses` list, to stay score-exact.
- Per-apiCall `.//value` scans replicate the back-end's nested-step double-counting.
- Found and fixed a latent crash: fast-xml-parser parses numeric tag text (e.g. epoch-millis dates) to `number`, so reading `#text` now coerces to string before `.trim()`.

## Jira
https://provartesting.atlassian.net/browse/PDX-508

## Test plan
- [x] yarn compile passes
- [x] full unit suite passes (1383 passing, +22 new Tier 2 tests)
- [x] mcp-smoke.cjs passes (60/60)
- [x] yarn lint passes
- [x] prettier clean

## Changes
- `src/mcp/tools/bestPracticesEngine.ts`: eight new validators + shared helpers (`collectElementsByTag`, `getStepValueElements`, `nodeText`, …), registered in `VALIDATOR_REGISTRY`.
- `test/unit/mcp/bestPracticesEngine.test.ts`: 22 parity tests (fire + pass + key edge cases per check type).
- `docs/mcp.md`: "Render / load-blocking rules" subsection documenting all eight.

> No customer-facing public-doc behaviour change beyond the documented rules; no new MCP tool/error code, so smoke `TOTAL_EXPECTED` is unchanged.

### CODE REVIEW

File: src/mcp/tools/bestPracticesEngine.ts [hunk 1]
```diff
@@ -1210,6 +1210,402 @@ function validateMustContainArgument(tc: XmlNode, rule: BPRule): BPViolation | n
...
+  let first: { ctx: ReturnType<typeof stepContext>; elem: string; context?: string } | null = null;
+  for (const call of getAllApiCalls(tc)) {
+    if (call['@_apiId'] !== SETVALUES_API_ID) continue;
+    const ctx = stepContext(call);
+    for (const elem of invalidElements) {
+      if (collectElementsByTag(call, elem).length) {
+        count++;
+        first ??= { ctx, elem };
+      }
+    }
```
[the remark this answers is not in the document]
Comment (mrdailey99) [reply src/mcp/tools/bestPracticesEngine.ts:1591]: Thanks — I checked this against the source-of-truth back-end (`SetValuesInvalidElementsValidator` in `quality-hub-agents` `best_practices_engine.py`), and the per-type counting here is intentional and required for score parity.

The back-end's first loop appends **one** violation entry per invalid element *type*, guarded by `if found:` — `len(found)` is stashed in the dict but is never used to multiply the entry count:

```python
for invalid_elem in self.INVALID_ELEMENTS:
    found = step.findall(f'.//{invalid_elem}')
    if found:
        violations.append({... 'count': len(found)})   # one entry per TYPE
```

The final `count` is `len(violations) if len(violations) > 1 else None`, i.e. the number of *entries*. So the type-level `count++` here matches the back-end exactly. The `namedValues` child loop below it is deliberately per-*instance* because the back-end's second loop (`for child in named_values: violations.append(...)`) appends one entry per child instance. The asymmetry you spotted is real, but it mirrors the back-end's own two loops — making both per-instance would over-count the `<namedValueSet>`/`<name>` part and *break* parity with the Lambda score.

Leaving as-is. Good catch to flag count semantics here, though — that's exactly where parity bugs hide.
